src/snkit/network.py

```diff
@@ -660,6 +647,36 @@ def line_endpoints(line):
     return start, end
 
 
+def intersection_endpoints(geom, output=[]):
```

Comment:
This is an unpleasant Python thing, especially if you've not seen it before: if you define a mutable default argument, it gets defined at the outer scope, so, for a list, repeated calls to the function will keep appending to the same list.

```python
>>> def append_to_default_list(val, list_=[]):
...     list_.append(val)
...     return list_
...
>>> append_to_default_list(1)
[1]
>>> append_to_default_list(2)
[1, 2]
>>> append_to_default_list(3)
[1, 2, 3]
```

To get the expected behaviour, I would suggest default `None`, then setting up the "real" default value inside the scope of the function:

```python
def intersection_endpoints(geom, output=None):
    ...
    if output = None:
        output = []
    ...
```